### PR TITLE
add tests and secrets-timeout

### PR DESCRIPTION
**Pull request by arjundashrath**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -10,23 +10,29 @@ interface HttpBinData {
...
+
+  if (slackWebhookUrl !== undefined && slackWebhookUrl !== ""){
     await sendToSlack(slackWebhookUrl, secretUrl);
   }
 
   var authIDToken = await core.getIDToken();
 
   var secretsString = core.getMultilineInput("secrets");
+  
   console.log(JSON.stringify(secretsString));
```
Comment (arjundashrath) [inline src/index.ts:36]: @varunsh-coder why are we logging the secretsString? seems redundant to me

**Comment by arjundashrath** [comment]
@varunsh-coder I'm facing an issue in the test.yml workflow. The jest tests are passing locally but not passing on the runner. Any suggestions?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      - main # to update code coverage
+
+permissions: # added using https://github.com/step-security/secure-workflows
+  contents: read
+concurrency:
+  group: ${{ github.workflow }}
+jobs:
+  test:
+    permissions:
+      id-token: write
```
Comment (varunsh-coder) [inline .github/workflows/test.yml:17]: Please remove `id-token` from the unit test workflow

**Comment by arjundashrath** [comment]
@varunsh-coder this test workflow is running successfully on my fork, but not on the main repo, it's because of the id-token permission. Any suggestions?

**Comment by varunsh-coder** [comment]
@arjundashrath the test is passing now. Can you please add the test for `setSecrets` method? Also, see if you can move more code into `common.ts` and add tests for it. 

Once this is done, please rename the `secrets-timeout` to `wait-timeout`.

Then test the changes using `wait-for-secrets@commitSHA` in a test repo to see if the changes are working fine.

**Comment by arjundashrath** [comment]
@varunsh-coder made the requested changes. Tested it as well, everything works as expected. Could not find anymore code to add to common.ts. The timeout feature is also working as expected

File: package.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
     "main": "index.js",
     "scripts": {
         "build": "ncc build src/index.ts --source-map",
-        "lint": "eslint src/**/*.ts"
+        "lint": "eslint src/**/*.ts",
+        "test": "jest --runInBand --detectOpenHandles --forceExit"
```
Comment (varunsh-coder) [inline package.json:9]: Do we still need the arguments?

File: src/index.ts [hunk 4]
```diff
@@ -10,24 +14,34 @@ interface HttpBinData {
...
+
+  var secretUrl = generateSecretURL(
+    environmentData[0],
+    environmentData[1],
+    environmentData[2]
+  );
 
   var slackWebhookUrl = core.getInput("slack-webhook-url");
+
+  var secretsTimeOut: number = +core.getInput("wait-timeout");
```
Comment (varunsh-coder) [inline src/index.ts:36]: Is there a core.getIntegerInput or something like that?

File: package.json (hunk 3 above)
Comment (arjundashrath) [reply package.json:9]: @varunsh-coder these arguments are not necessary but they are still helpful to add to tests. Their usage does not affect the performance of the test. the `runinband` arg makes sure the tests run serially that helps in easier debugging, and the other two args make sure that the test does not run indefinitely and describes where the test went into the indefinite process. The args provide future proofing when more tests will be introduced so I think its better to have them in place.

File: src/index.ts (hunk 4 above)
Comment (arjundashrath) [reply src/index.ts:36]: @varunsh-coder no, there is only `core.getInput` and `core.getBooleanInput`

**Review by varunsh-coder — APPROVED** [review]

**Pull request merged**